src/main/scala/fm/http/client/Response.scala

```diff
@@ -54,10 +54,13 @@ sealed abstract class Response(response: HttpResponse) extends Closeable {
     import Response.CharsetRegex
 
     response.headers().get("Content-Type").toBlankOption.flatMap { contentType: String =>
-      for {
-        CharsetRegex(charset) <- CharsetRegex.findFirstIn(contentType)
-        if charsetIsSupported(charset)
-      } yield Charset.forName(charset)
+      MimeTypes.mimeTypeToCharset.get(contentType) orElse
+      (
+        for {
+          CharsetRegex(charset) <- CharsetRegex.findFirstIn(contentType)
+          if charsetIsSupported(charset)
+        } yield Charset.forName(charset)
+      )
```

Comment:
This logic doesn't seem very flexible.

Is `application/json; charset=ISO-8859-1` valid?  That doesn't work with this code.

What about other `Content-Type`s where we want to be able to set a default but the explicit `charset=...` should still work?

Reply:
makes sense, I'll swap the ordering add add more tests